### PR TITLE
ci(NODE-5090): download node to local directory

### PR DESCRIPTION
**Pull request by nbbeeken**
### Description

Backport #3583

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/install-dependencies.sh [hunk 1]
```diff
@@ -1,108 +1,98 @@
...
-NODE_VERSION=$(latest_version_for_node_major $NODE_MAJOR_VERSION)
-NODE_VERSION=${NODE_VERSION:1} # :1 gets rid of the leading 'v'
-
-echo "set version to $NODE_VERSION"
-
-# output node version to expansions file for use in subsequent scripts
-cat <<EOT > deps-expansion.yml
-  NODE_VERSION: "$NODE_VERSION"
-EOT
+NODE_LTS_NAME=${NODE_LTS_NAME:-erbium}
```
Comment (nbbeeken) [inline .evergreen/install-dependencies.sh:4]: default on this branch is 12

File: .evergreen/install-dependencies.sh [hunk 2]
```diff
@@ -1,108 +1,98 @@
...
+cat <<EOT > .npmrc
+prefix=$NODE_ARTIFACTS_PATH/npm_global
+EOT
+
+# Cannot upgrade npm version for node 12
+if [[ $operating_system != "win" ]] && [[ $NODE_LTS_NAME != "erbium" ]]; then
+  # Update npm to latest when we can
+  npm install --global npm@latest
+  hash -r
+fi
```
Comment (nbbeeken) [inline .evergreen/install-dependencies.sh:94]: Had to exempt 12 from upgrading npm, should we try to upgrade to npm v8 instead?

Comment (durran) [reply .evergreen/install-dependencies.sh:94]: Seems like that would be a good idea. What's it on 12 by default? 6.x?

Comment (nbbeeken) [reply .evergreen/install-dependencies.sh:94]: Yep 12.9.1 ships with 6.10.2, but it should be able to run up to 8

I'll update this to install `@8` just for node 12 tasks

https://www.npmjs.com/package/npm?activeTab=versions

**Comment by nbbeeken** [comment]
evergreen retry

**Review by durran — DISMISSED** [review]

**Pull request merged**